.gitlab-ci.yml

```diff
@@ -59,7 +59,10 @@ stages:
 
 .on_push:
   rules:
-    - if: $CI_PIPELINE_SOURCE != "schedule"
+    - if: $CI_MERGE_REQUEST_IID
+    - if: $CI_OPEN_MERGE_REQUESTS
+      when: never
```

Comment:
is the `when: never` a debugging bit? If it is truly never, we can just remove this section right?
```suggestion
```

Reply:
it is not unfortunately

Reply:
Sorry for the brevity. To expand, this is to support running the test pipeline on both branches and merge requests, but not both if the branch has an open merge request.

It's a weird and complicated set of rules taken from https://gitlab.com/gitlab-org/gitlab/-/issues/201845#note_591075909 to enable that behavior.